source/dbo/Stored Procedures/Unmatched Specimens/uspGenerateUnmatchedSpecimensLinkedNotifications.sql

```diff
@@ -0,0 +1,66 @@
+﻿CREATE PROCEDURE [dbo].[uspGenerateUnmatchedSpecimensLinkedNotifications]
+AS
+BEGIN TRY
+
+	-- First populate RejectSpecimenUsers table
+	EXEC dbo.uspGenerateRejectSpecimenUsers
+
+	TRUNCATE TABLE [dbo].[UnmatchedSpecimensLinkedNotifications]
+	
+	SELECT us.ReferenceLaboratoryNumber, nsm.NotificationID, 'Unmatch' AS EventType INTO #PreviousMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.NotificationSpecimenMatch nsm ON nsm.ReferenceLaboratoryNumber = us.ReferenceLaboratoryNumber
+	WHERE nsm.MatchType = 'Rejected'
+
+	SELECT us.ReferenceLaboratoryNumber, nsm.NotificationID, 'RejectPotentialMatch' AS EventType INTO #PreviousPossibleMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.NotificationSpecimenMatch nsm ON nsm.ReferenceLaboratoryNumber = us.ReferenceLaboratoryNumber
+	WHERE nsm.MatchType = 'Rejected-Possible'
+
+	SELECT mr.clientsourceID AS ReferenceLaboratoryNumber, mr.rclientsourceID AS NotificationID, 'PoorQualityMatch' AS EventType INTO #PoorQualityMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.MatchResults mr ON mr.clientsourceID = us.ReferenceLaboratoryNumber
+	--Exclude any matches that have previously been rejected (otherwise every rejection has an accompanying match)
+	LEFT JOIN #PreviousMatch pm on pm.NotificationID = mr.rclientsourceID AND pm.ReferenceLaboratoryNumber = mr.clientsourceID
+	LEFT JOIN #PreviousPossibleMatch ppm on ppm.NotificationID = mr.rclientsourceID AND ppm.ReferenceLaboratoryNumber = mr.clientsourceID
+	WHERE pm.EventType IS NULL AND ppm.EventType IS NULL
+	AND mr.[weight] < 34.50
```

Comment:
I think the logic around poor quality matches is more complicated than this (there are multiple confirmed matches with weights in the 20's) but I think this will do

Reply:
It's the value of 'AutoMatch' from the MatchingConfiguration table in the SpecimenMatching db

Reply:
In that case we should fetch the value directly from the table - we shouldn't be expecting someone who changes the configured value to have to remember to change this too!